simulation/src/curp_group.rs

```diff
@@ -257,13 +257,13 @@ impl CurpGroup {
     }
 
     pub async fn get_leader(&self) -> (ServerId, u64) {
-        for _ in 0..5 {
+        loop {
```

Comment:
We should add a timeout mechanism to this loop to prevent endless waiting in some rare cases.

Reply:
As the time is simulated, it's hard to give a specific timeout for the waiting. In my testing, even a long duraion of waiting may fail to get the leader.

It seems the client will always get the leader, as long as the cluster is able to conduct a successful election?